argo/providers/base/provider.go

```diff
@@ -0,0 +1,22 @@
+package base
+
+import (
+	"context"
+)
+
+type PipelineConfig struct {
```

Comment:
Should we rename this something like `PipelineDefinition`? It would really read terribly as `RunConfigurationConfig` when we start implementing it 😓